pycheribuild/projects/project.py

```diff
@@ -2499,25 +2529,25 @@ def process(self):
                         return
                     self._force_clean = True
 
-        last_clean_build_date_path = Path(self.buildDir, ".last_clean_build_date")
-        if self.full_rebuild_if_older_than is not None and not self.config.clean:
-            last_clean_date = None
+        required_clean_counter = self._parse_require_clean_build_counter()
+        clean_counter_in_build_dir = None  # typing.Optional[int]
+        last_clean_counter_path = self._last_clean_counter_path()
+        if required_clean_counter is not None:
+            # Check if the last clean build had a smaller counter than the current required on and if so perform a clean
+            # build and increment the value in the build directory.
             try:
-                if last_clean_build_date_path.exists():
-                    # Note: fromisoformat() only available in python 3.7
-                    # last_clean_date = datetime.datetime.fromisoformat(last_clean_build_date_path.read_text().strip())
-                    last_clean_date = datetime.datetime.strptime(last_clean_build_date_path.read_text().strip(), self._LAST_CLEAN_TIME_FMT)
-                    # Update timezone to allow equality comparison with utcnow()
-                    last_clean_date = last_clean_date.replace(tzinfo=datetime.timezone.utc)
-            except ValueError as e:
-                self.warning("Could not parse contents of", last_clean_build_date_path, e)
-            if last_clean_date is None:
-                self._force_clean = True
-                self.info("Forcing full rebuild since", last_clean_build_date_path, "could not be parsed")
-            elif last_clean_date < self.full_rebuild_if_older_than:
+                clean_counter_in_build_dir = int(last_clean_counter_path.read_text().strip())
+                if clean_counter_in_build_dir < required_clean_counter:
+                    self.info("Forcing full rebuild since clean counter in build dir (", clean_counter_in_build_dir,
+                        ") is less than required minimum ", required_clean_counter, sep="")
+                    self._force_clean = True
+                else:
+                    self.verbose_print("Not forcing clean build since clean counter in build dir",
+                        clean_counter_in_build_dir, "is >= required minimum", required_clean_counter)
+            except Exception as e:
+                self.warning("Could not parse", last_clean_counter_path, "-> assuming clean build is required.", e)
```

Comment:
We probably don't want to print a warning if the file doesn't exist; that should be handled gracefully as if the counter is -INF.

Reply:
Yeah I just had that warning there to see that it did the right thing without a debugger.
 Will fix